src/invariant/BucketListIsConsistentWithDatabaseTests.cpp

```diff
@@ -278,7 +278,7 @@ class ApplyBucketsWorkAddEntry : public ApplyBucketsWork
 
     ~ApplyBucketsWorkAddEntry()
     {
-        REQUIRE(mAdded);
+        // REQUIRE(mAdded);
```

Comment:
The only issue with this REQUIRE (and the corresponding ones for Delete and Modify) is the possibility of throwing from a constructor, right? These REQUIRE statements are used to check that the test actually did make the required alteration to the data, so I consider them too important to simply comment out.

The solution that I propose is to check this after `ApplyBucketsWork::onSuccess` returns `WORK_SUCCESS`. For example, we could make the following change at line 341 (and corresponding changes at lines 386 and 498):
```
auto abwRes = ApplyBucketsWork::onSuccess();
if (abwRes == WORK_SUCCESS)
{
    REQUIRE(mAdded);
}
return abwRes;
```
It would be even better from a code duplication perspective to make a function out of the above code snippet since it will be used 3 times.

This change would still cause the REQUIRE statements to occur after ApplyBucketsWork is complete, but would avoid putting the REQUIRE in the constructor. In fact, at this point we could simply use the default destructor instead of providing our own. Let me know if you see any other issues with this solution.

Reply:
I moved the checks per your suggestion. I didn't create a function for this though as it seems overkill.